test/lit/source-map.wast

```diff
@@ -26,8 +26,9 @@
 )
 
 ;; CHECK:  (func $foo (param $x i32) (param $y i32)
-;; CHECK-NEXT:   ;;@ src.cpp:20:1
```

Comment:
I'm not very familiar with this part of the code. A few questions:
- Why didn't we print the debug location for control flow structure before?
- It looks this `src.cpp:20:1` is the debug info for not `if` but `i32.add`, given that we didn't print debug info for control flow structures. Then why was it printed before `if` and not before `i32.add`?

Reply:
Unfortunately I don't think anyone is very familiar with it :smile: It was written a long time ago (and not by me).

I can't see a reason for not printing it for control flow structures before. That might have just been an omission.

I'm not sure about the second question. I know there are some heuristics for how to print and emit stuff, that involve looking at recent instructions or such, so probably related to that?

Overall, my approach here considering this unfamiliar code is to get more tests on it, and improve on them. This PR seems to do that as after it we appear to perfectly roundtrip the input.